src/client/stdio.ts

```diff
@@ -158,6 +149,15 @@ export class StdioClientTransport implements Transport {
       this._process.stdout?.on("error", (error) => {
         this.onerror?.(error);
       });
+
+      this._process.stderr?.on("data", (chunk) => {
+        this._readBuffer.append(chunk);
+        this.processReadBuffer();
```

Comment:
This is incorrect—it will push stderr logging into protocol communication, causing errors.